src/operator/skipWhile.ts

```diff
@@ -38,17 +36,21 @@ class SkipWhileSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const destination = this.destination;
-    if (this.skipping === true) {
-      const index = this.index++;
-      const result = tryCatch(this.predicate)(value, index);
-      if (result === errorObject) {
-        destination.error(errorObject.e);
-      } else {
-        this.skipping = Boolean(result);
-      }
+    if (this.skipping) {
+      this.tryCallPredicate(value);
     }
-    if (this.skipping === false) {
+
+    if (!this.skipping) {
```

Comment:
Actually, you can't really return 'skipping', so it's probably better to either have it in the same function, or move it to it's own function. What do you think?